### ISSUE TITLE
Problem: zmq_msg_send doc incorrect for large messages.

### ISSUE THREAD

**Issue by evoskuil** [zeromq/libzmq#2488]
Documentation for [zmq_msg_send](http://api.zeromq.org/4-2:zmq-msg-send) states:

### "Return value
The zmq_msg_send() function shall return number of bytes in the message if successful. Otherwise it shall return -1 and set errno to one of the values defined below."

However due to the implementation:
```cpp
s_sendmsg (zmq::socket_base_t *s_, zmq_msg_t *msg_, int flags_)
{
    size_t sz = zmq_msg_size (msg_);
    int rc = s_->send ((zmq::msg_t *) msg_, flags_);
    if (unlikely (rc < 0))
        return -1;

    size_t max_msgsz = INT_MAX;
    return (int) (sz < max_msgsz? sz: max_msgsz);
}
```
the documentation is incorrect for any message of size that exceeds `INT_MAX`. Either the documentation should be changed to indicate that the return is set to `INT_MAX` in the case of success for such messages, or the implementation should be changed to limit message size accordingly. Otherwise behavior in these cases is ambiguous to the API caller.

**Comment by bluca** [comment]
The documentation should be fixed I think, the problem if I remember correctly is that the value would just be truncated if higher than INT_MAX given the API returns an int.
I sincerely hope nobody ever has to send 2 gigabytes of data in a single message though :-)

**Comment by bluca** [comment]
Fixed by #3266

**Comment by akropp** [comment]
> The documentation should be fixed I think, the problem if I remember correctly is that the value would just be truncated if higher than INT_MAX given the API returns an int. I sincerely hope nobody ever has to send 2 gigabytes of data in a single message though :-)

Actually, I just tried to send more than 2gb of data and hit the assertion in zmq.hpp that checks whether the return value from zmq_msg_recv matches the size of the message received.
